SteamKit2/SteamKit2/Networking/Steam3/UdpConnection.cs

```diff
@@ -356,22 +366,49 @@ private bool DispatchMessage()
         /// <summary>
         /// Processes incoming packets, maintains connection consistency, and oversees outgoing packets.
         /// </summary>
-        private void NetLoop()
+        private void NetLoop(object param)
         {
             // Variables that will be used deeper in the function; locating them here avoids recreating
             // them since they don't need to be.
-            EndPoint packetSender = (EndPoint) new IPEndPoint(IPAddress.Any, 0);
+            var userRequestedDisconnect = false;
+            EndPoint packetSender = (EndPoint)new IPEndPoint(IPAddress.Any, 0);
             byte[] buf = new byte[2048];
 
-            timeOut = DateTime.Now.AddSeconds(TIMEOUT_DELAY);
-            nextResend = DateTime.Now.AddSeconds(RESEND_DELAY);
+            if (param is IPEndPoint)

```

Comment:
This block seems iffy. You're doing `is` then `as` (better to just do `as` and check for `null`), followed by `as` and dereference. If `param` is neither an `IPEndPoint` not a `Task<IPEndPoint>`, this will just be a `NullReferenceException`. Better to just do an `as` for `IPEndPoint`, then a cast for `Task<IPEndPoint>`. Or ditch `IPEndPoint` entirely and always supply a task.